x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClustersIT.java

```diff
@@ -127,8 +127,10 @@ void indexDocs(RestClient client, String index, List<Doc> docs) throws IOExcepti
         refresh(client, index);
     }
 
-    private Map<String, Object> run(String query) throws IOException {
-        Map<String, Object> resp = runEsql(new RestEsqlTestCase.RequestObjectBuilder().query(query).build());
+    private Map<String, Object> run(String query, boolean includeCCSMetadata) throws IOException {
+        Map<String, Object> resp = runEsql(
+            new RestEsqlTestCase.RequestObjectBuilder().query(query).includeCCSMetadata(includeCCSMetadata).build()
```

Comment:
Using `include_ccs_metadata` only on this IT test suite is not enough I think.
Users can use `include_ccs_metadata` on any kind of REST request (multi cluster, single cluster single node, multi-node, mixed version, json/txt/tsv formats). I would, also, double check that `"columnar": true` (see [this one in docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-query-api.html#esql-query-api-request-body)) still works with this new addition to the JSON response.

Reply:
> I would, also, double check that `"columnar": true` still works with this new addition to the JSON response.

It does. I tested it manually and I have added a test to MultiClustersIT that uses columnar format with `include_ccs_metadata=true` and it is passing.

> I think include_ccs_metadata parameter is of the same type: it can only be used for json format and I think we need a similar check

OK, I can add this restriction. If later we want to support it with SMILE, YAML and CBOR we can relax it.

Reply:
I have added a restriction that `include_ccs_metadata` can only be used with format=JSON. Tests have been added / updated to match this restriction.